modules/replicator/src/main/java/org/apache/ignite/internal/replicator/Replica.java

```diff
@@ -17,126 +17,22 @@
-
-    /** Topology aware Raft client. */
-    private final TopologyAwareRaftGroupService raftClient;
-
-    /** Instance of the local node. */
-    private final ClusterNode localNode;
-
-    // TODO IGNITE-19120 after replica inoperability logic is introduced, this future should be replaced with something like
-    //     VersionedValue (so that PlacementDriverMessages would wait for new leader election)
-    /** Completes when leader is elected. */
-    private final CompletableFuture<AtomicReference<ClusterNode>> leaderFuture = new CompletableFuture<>();
-
-    /** Container of the elected leader. */
-    private final AtomicReference<ClusterNode> leaderRef = new AtomicReference<>();
-
-    /** Latest lease expiration time. */
-    private volatile HybridTimestamp leaseExpirationTime;
-
-    /** External executor. */
-    // TODO: IGNITE-20063 Maybe get rid of it
-    private final ExecutorService executor;
-
-    private final PlacementDriver placementDriver;
-
-    private final ClockService clockService;
-
-    private final BiFunction<ReplicationGroupId, HybridTimestamp, Boolean> replicaReservationClosure;
-
+public interface Replica {
     /**
-     * The constructor of a replica server.
+     * Returns replica's RAFT client.
      *
-     * @param replicaGrpId Replication group id.
-     * @param listener Replica listener.
-     * @param storageIndexTracker Storage index tracker.
-     * @param localNode Instance of the local node.
-     * @param executor External executor.
-     * @param placementDriver Placement driver.
-     * @param clockService Clock service.
-     * @param replicaReservationClosure Closure that will be called to reserve the replica for becoming primary. It returns whether
-     *     the reservation was successful.
+     * @return RAFT client.
      */
-    public Replica(
-            ReplicationGroupId replicaGrpId,
-            ReplicaListener listener,
-            PendingComparableValuesTracker<Long, Void> storageIndexTracker,
-            ClusterNode localNode,
-            ExecutorService executor,
-            PlacementDriver placementDriver,
-            ClockService clockService,
-            BiFunction<ReplicationGroupId, HybridTimestamp, Boolean> replicaReservationClosure
-    ) {
-        this.replicaGrpId = replicaGrpId;
-        this.listener = listener;
-        this.storageIndexTracker = storageIndexTracker;
-        this.raftClient = raftClient();
-        this.localNode = localNode;
-        this.executor = executor;
-        this.placementDriver = placementDriver;
-        this.clockService = clockService;
-        this.replicaReservationClosure = replicaReservationClosure;
-
-        raftClient.subscribeLeader(this::onLeaderElected);
-    }
-
-    /** Returns Raft-client. */
-    public final TopologyAwareRaftGroupService raftClient() {
-        return (TopologyAwareRaftGroupService) listener.raftClient();
-    }
+    TopologyAwareRaftGroupService raftClient();
```

Comment:
I'd expect given method to be @Deprecated(forRemoval=true) with todo linked to some of @JAkutenshi tickets.

Reply:
done